src/ucp/wireup/wireup.c

```diff
@@ -1033,7 +1027,8 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep,
     ucs_assert(old_key->wireup_msg_lane != UCP_NULL_LANE);
     /* wireup lane hasn't been selected by the new configuration: only this
      * function should select it */
-    ucs_assert(new_key->wireup_msg_lane == UCP_NULL_LANE);
+    ucs_assert((new_key->wireup_msg_lane == UCP_NULL_LANE) ||
```

Comment:
0. why we need this check? is `wireup_msg_lane` selected for the new key despite on the checks in https://github.com/openucx/ucx/blob/7a452bc101a73f147718e7465cdf38b6d71da6bd/src/ucp/wireup/select.c#L1660
1. better to not compare lane indices. can use `ucp_ep_config_find_match_lane(new_key, new_key->wireup_msg_lane, old_key) == old_key->wireup_msg_lane` condition instead.
2. need to fix the comment above this assertion.

Reply:
> why we need this check?

hmm... seems like somebody unset `UCP_EP_FLAG_LOCAL_CONNECTED` flag or there is a mix up with tmp ep, will try to figure out this.